### PR TITLE
Wireshark3: add python3.10 variant

### PR DESCRIPTION
**Pull request by jrabinow**
#### Description

wireshark3: add python310 variant

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 12.2.1 21D62 x86_64
Xcode 13.1 13A1030d

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: net/wireshark3/Portfile [hunk 1]
```diff
@@ -167,22 +168,37 @@ variant chmodbpf description {Enable Wireshark to access macOS capture devices}
...
 
-variant python36 conflicts python37 python38 python39 description {Use python36 during build} {
-    depends_build-append    port:python36
-}
+set pythons_suffixes {36 37 38 39}
 
-variant python37 conflicts python36 python38 python39 description {Use python37 during build} {
-    depends_build-append    port:python37
+set pythons_ports {}
+foreach s ${pythons_suffixes} {
```
Comment (ghosthound) [inline net/wireshark3/Portfile:174]: Please use more descriptive variable names, like "pySuffix" (here), "pyVersion" below, that sort of thing.

Comment (jrabinow) [reply net/wireshark3/Portfile]: Done. I did my best to use the same variable names as in the rest of macports (`py_ver`, `py_ver_nodot`) and build from there

Comment (ghosthound) [reply net/wireshark3/Portfile]: Cool, thanks, and great idea on using the same var names used elsewhere.

File: net/wireshark3/Portfile [hunk 2]
```diff
@@ -167,22 +168,37 @@ variant chmodbpf description {Enable Wireshark to access macOS capture devices}
...
 
-variant python39 conflicts python36 python37 python38 description {Use python39 during build} {
-    depends_build-append    port:python39
+# make sure one of the +python* variants is set
+set python_isset false
+foreach py_ver_nodot ${pythons_suffixes} {
+    if {!${python_isset}} {
+        set python_isset  [variant_isset python${py_ver_nodot}]
+    }
+}
```
Comment (ghosthound) [inline net/wireshark3/Portfile:194]: While it doesn't really matter much here as the "pythons_suffixes" is so short and I expect the common case is to not provide a python variant, I generally prefer to bail out of the loop as soon as we find that a python variant is set, like:
```foreach py_ver_nodot ${pythons_suffixes} {
    if { [variant_isset python${py_ver_nodot}] } {
        set python_isset true
        break
    }  
}
```
Thoughts?

Comment (jrabinow) [reply net/wireshark3/Portfile:195]: Sure the list is short, but `break` can't hurt, and it makes for cleaner code overall. I just edited in your suggestion :-)

**Comment by ghosthound** [comment]
Looks good, thanks for the help.

**Pull request merged**